test/corpus/functions.txt

```diff
@@ -1003,3 +1003,166 @@ $function$
           (identifier))
         (keyword_end)
         (dollar_quote)))))
+
+================================================================================
+Function body with a $$ string literal
+================================================================================
+
+create or replace function public.do_stuff()
+  returns trigger
+  language plpgsql
+as $a$
+begin
+
+  -- comment!
+  with knn as (
+    select
+      h.alpha,
+      -- TODO factor in distance
+      avg(e.beta) as e_beta
+    from htable h
+    cross join lateral (
+      select
+        id,
+        gamma,
+        delta,
+        centroid
+      from ftable
+      limit 3
+    ) as e
+    group by h.alpha
+  )
+  update htable set epsilon = epsilon + e_beta
+  from knn
+  where knn.alpha = htable.alpha;
+
+  return $a$test$a$;
```

Comment:
```suggestion
  return $b$test$b$;
```
reusing `a` errors:
```
ERROR:  syntax error at or near "test$a$"
LINE 29:   return $a$test$a$;
```

Reply:
Maybe there is `make compile` to do before `make test`